src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ThurloeDAO.scala

```diff
@@ -22,11 +22,44 @@ trait ThurloeDAO extends LazyLogging with ReportsSubsystemStatus {
   def getProfile(userInfo: UserInfo): Future[Option[Profile]]
   def getAllUserValuesForKey(key: String): Future[Map[String, String]]
   def saveProfile(userInfo: UserInfo, profile: BasicProfile): Future[Unit]
+
+  /**
+    * Save KVPs for myself - the KVPs will be saved to the same user that authenticates the call.
+    * @param userInfo contains the userid for which to save KVPs and that user's auth token
+    * @param keyValues the KVPs to save
+    * @return success/failure of save
+    */
   def saveKeyValues(userInfo: UserInfo, keyValues: Map[String, String]): Future[Try[Unit]]
 
+  /**
+    * Save KVPs for a different user - the KVPs will be saved to the "forUserId" user,
+    * but the call to Thurloe will be authenticated as the "callerToken" user.
+    * @param forUserId the userid of the user for which to save KVPs
+    * @param callerToken auth token of the user making the call
+    * @param keyValues the KVPs to save
+    * @return success/failure of save
+    */
+  def saveKeyValues(forUserId: String, callerToken: WithAccessToken, keyValues: Map[String, String]): Future[Try[Unit]]
```

Comment:
It'd be good to type `forUserId` more strongly via something like a value class:

```scala
final case class UserId(value: String) extends AnyVal
```